cypress/support/pages/rooms/pageRooms.js

```diff
@@ -114,20 +115,60 @@ class Rooms {
 		});
 	}
 
-	verifyRoomDeletion(roomName) {
+	verifyRoomDeletion(roomNamePrefix) {
 		cy.get("body").then(($body) => {
-			if (!$body.text().includes(roomName)) {
-				cy.log(`All rooms with name "${roomName}" deleted successfully.`);
-			} else {
-				cy.get(Rooms.#roomTitle).should("not.contain.text", roomName);
-			}
+			expect(
+				$body.text().includes(roomNamePrefix),
+				`Rooms containing "${roomNamePrefix}" should not exist`
+			).to.be.false;
 		});
 	}
 
-	deleteAllRoomsWithName(roomName) {
-		cy.wait(2000);
-		this.deleteElementsWithText(Rooms.#roomTitle, roomName);
-		this.verifyRoomDeletion(roomName);
+	deleteAllRoomsWithName(roomNamePrefix) {
+		cy.wait(1000);
+		const deleteNext = () => {
+			// wait until room titles are rendered (or not)
+			cy.get("body").then(($body) => {
+				const titles = $body.find('[data-testid^="room--title-"]');
+
+				// find any room that still contains the prefix
+				const room = [...titles].find((el) =>
+					el.innerText.includes(roomNamePrefix)
+				);
+
+				// nothing left → verify and stop
+				if (!room) {
+					this.verifyRoomDeletion(roomNamePrefix);
+					return;
+				}
+
+				// extract index from data-testid="room--title-{index}"
+				const index = room
+					.getAttribute("data-testid")
+					.replace("room--title-", "");
+
+				// open and delete the room
+				cy.get(`[data-testid="room-open-button-${index}"]`)
+					.should("be.visible")
+					.click();
+
+				cy.get(Rooms.#roomDetailFAB).should("be.visible").click();
+				cy.get(Rooms.#btnRoomDelete).should("be.visible").click();
+				cy.get(Rooms.#deletionConfirmationModalTitle).should("exist");
+				cy.get(Rooms.#confirmButtonOnModal).should("be.visible").click();
+
+				// IMPORTANT: wait for navigation back + list render
+				cy.wait(1500);
+
+				// retry after UI is ready
+				cy.reload().then(() => {
```

Comment:
it would be enough to have like this don't need `then` block
 
```js
cy.reload()
deleteNext();
```

Reply:
as cypress commands are asynchronous and queued, so its safer option that deleteNext() will run immediately, before cy.reload() finishes (using then()).